packages/web-components/src/label/label.base.ts

```diff
@@ -0,0 +1,115 @@
+import { attr, FASTElement, observable } from '@microsoft/fast-element';
+import { labelElementTemplate } from './label.template.js';
+
+/**
+ * The base class used for constructing a fluent-label custom element
+ *
+ * @tag fluent-label
+ *
+ * @public
+ */
+export class BaseLabel extends FASTElement {
+  /**
+   * 	Specifies styles for label when associated input is disabled
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `disabled`
+   */
+  @attr({ mode: 'boolean' })
+  public disabled: boolean = false;
+
+  /**
+   * The label's for attribute.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `for`
+   */
+  @attr({ attribute: 'for' })
+  public htmlFor!: string;
+
+  /**
+   * 	Specifies styles for label when associated input is a required field
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `required`
+   */
+  @attr({ mode: 'boolean' })
+  public required: boolean = false;
+
+  /**
+   * The default slotted content.
+   *
+   * @public
+   */
+  @observable
+  defaultSlottedContent!: Node[];
+
+  /**
+   * Ensures that a label element is present in the light DOM. It may be a child or parent of this element.
+   *
+   * @param prev - The previous list of slotted content
+   * @param next - The current list of slotted content
+   * @public
+   *
+   * @remarks
+   * If a `<label>` element is not present when connected, one will be created and inserted into the light DOM, and all
+   * slotted content will be moved into the element.
+   */
+  defaultSlottedContentChanged(prev: Node[] | undefined, next: Node[] | undefined): void {
+    const labelElement =
+      this.closest('label') ??
+      next?.find<HTMLLabelElement>((x): x is HTMLLabelElement => x instanceof HTMLLabelElement);
+
+    if (labelElement) {
+      this.labelElement = labelElement;
+      return;
+    }
+
+    if (next?.length) {
+      this.insertLabel();
+    }
+  }
```

Comment:
Should we _perhaps_...note in the documentation that we would suggest using `<field>` with `<label>` rather than this?

Reply:
I'll address and update this component's README since it's currently very inaccurate. The issue here is that there are too many possible approaches that are potentially being used in the wild, and the `<fluent-label>` component doesn't provide _any_ semantic meaning _or_ functionality beyond styling and adding a red asterisk when _it_ is marked as required.

Styling could be handled separately, with a `<fluent-text>`. From what I've seen, the red asterisk functionality is expected to "just work" and is the source of many accessibility concerns. I'm still not certain if the asterisk should be considered localizable content or a static glyph for sighted users, but since it already exists in the `<fluent-label>` component, this feels like the safest approach.